### PR TITLE
Add CITATION file with CHEP 2024 article

### PR DESCRIPTION
**Pull request by sethrj**
This adds a [Citation File Format](https://zenodo.org/records/5171937/preview/schema-guide.md) file with the (currently best) Celeritas reference. This [is rendered by github](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-citation-files) as the following:
github screenshot

### CODE REVIEW

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# Pleace cite the following article when referencing Celeritas in a publication:
```
Comment (amandalund) [inline CITATION.cff:1]: Are these comments left in intentionally?

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+#      pages = {11005},
+#      doi = {10.1051/epjconf/202429511005}
+#    }
+#
+# (CFF):
+
+cff-version: 1.2.0
+type: software
+message: "Please cite the following article when referencing Celeritas in a publication."
+title: "Celeritas: Accelerating Geant4 with GPUs"
```
Comment (amandalund) [inline CITATION.cff]: For the software citation, should we also include keys like `repository-code`, `doi`, `license`,`version`?

File: CITATION.cff [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
...eritas [1] is a new Monte Carlo (MC) detector simulation code designed for computationally intensive applications (specifically, High Luminosity Large Hadron Collider (HL-LHC) simulation) on high-performance heterogeneous architectures. In the past two years Celeritas has advanced from prototyping a GPU-based single physics model in infinite medium to implementing a full set of electromagnetic (EM) physics processes in complex geometries. The current release of Celeritas, version 0.3, has incorporated full device-based navigation, an event loop in the presence of magnetic fields, and detector hit scoring. New functionality incorporates a scheduler to offload electromagnetic physics to the GPU within a Geant4-driven simulation, enabling integration of Celeritas into high energy physics (HEP) experimental frameworks such as CMSSW. On the Summit supercomputer, Celeritas performs EM physics between 6× and 32× faster using the machine’s Nvidia GPUs compared to using only CPUs. When running a multithreaded Geant4 ATLAS test beam application with full hadronic physics, using Celeritas to accelerate the EM physics results in an overall simulation speedup of 1.8–2.3× on GPU and 1.2× on CPU.
```
Comment (amandalund) [inline CITATION.cff]: Should the abstract go with the article citation? Or shortened/updated if kept here? (Also remove the dangling citation `[1]`?)

File: CITATION.cff [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+preferred-citation:
```
Comment (amandalund) [inline CITATION.cff:70]: In the "Citing Celeritas" section in the README we point to the [DOECode](https://www.osti.gov/doecode/biblio/94866) registration. Do we want to make sure we're consistent with what we ask for there and our preferred citation here?

**Review by amandalund — APPROVED** [review]
Cool, thanks @sethrj!

File: CITATION.cff (hunk 4 above)
Comment (sethrj) [reply CITATION.cff:70]: Oh good point. I'll add the DOE code as the main "reference" and then update both to the "citable" reference.

File: CITATION.cff (hunk 3 above)
Comment (sethrj) [reply CITATION.cff]: I based the overall layout on spack's citation, but yes I think it makes more sense to put the abstract with the associated publication.

File: CITATION.cff (hunk 1 above)
Comment (sethrj) [reply CITATION.cff:1]: Yeah, it seems like some repositories present human readable versions as well.

Comment (sethrj) [reply CITATION.cff:1]: Yeah, it seems like some repositories present human readable versions as well (Spack).

**Pull request merged**